pyscf/neo/hf.py

```diff
@@ -378,17 +378,119 @@ def _build_eri(mol1, mol2, cart):
                                              size1, size1 + size2),
                                  aosym='s4')
 
+def cholesky_eneri(mol_n, auxmol_e, int3c, int2c, max_memory,
+                   aosym='s2ij', decompose_j2c='cd'):
```

Comment:
Apart from _eig_decompose lines, I don't see the point rewriting a function. Directly using df.incore.cholesky_eri should just work (if using 'cd'). Not a good practice to remove the additional function signatures and those timing codes.